DOCS-12973 DG semantic extensions (initial commit) (#58)
mulesoft/docs-anypoint-datagraph: modules/ROOT/pages/overview-cli-extensions.adoc

= Using Anypoint Platform DataGraph CLI and API Extensions

> You can use the Anypoint CLI DataGraph commands to add and edit API sources in DataGraph. To do so you must first add the appropriate DataGraph API extensions to the API source. 

Reviewer: ```suggestion
You can use the Anypoint CLI DataGraph commands to add and edit API sources in DataGraph. To do so, you must first add the appropriate DataGraph API extensions to the API source. 
```

== Adding Extensions to Your Project

> You must add the extensions to your API-specification project as a dependency, and then publish the project to Exchange. After you publish your API-specification project, you can add the API to DataGraph using the CLI.

Reviewer: ```suggestion
You must add the extensions to your API-specification project as a dependency and then publish the project to Exchange. After you publish your API-specification project, you can add the API to DataGraph using the CLI.
```

Use API Designer to add the extensions to your project. Follow the steps in xref:design-center::design-add-api-dependency.adoc#add-dependencies-to-an-api-specification-project[Add Dependencies to an API-Specification Project], choosing *Fragments* as the dependency type. 

API Designer adds the required dependencies to `exchange.json` and validates the syntax of any annotations you added before publishing your project to Exchange.

== Maintaining RAML Compatibility

> To maintain compatibility in RAML specifications with tools that do not support API extensions, these extensions includes a “RAML companion” that provides a simple version of RAML annotations definition. To use DataGraph API extensions in RAML, add the following code to the main file of the API-project specification:

Reviewer: ```suggestion
To maintain compatibility in RAML specifications with tools that do not support API extensions, these extensions include a “RAML companion” that provides a simple version of RAML annotations definition. To use DataGraph API extensions in RAML, add the following code to the main file of the API-project specification:
```

----
uses:
  dg: exchange_modules/<group>/<asset>/<version>/datagraph-api-extensions.raml 
----

> You can change the prefix `dg` if needed, but it must match the prefix you use the one use in the extensions.

Reviewer: ```suggestion
You can change the prefix `dg` if needed, but it must match the prefix used in the extensions.
```

The `<group>`, `<asset>`, and `<version>` reference the API extensions that are published in Exchange.

> == Using the DataGraph CLI vs. the DataGraph UI

Reviewer: ```suggestion
== Using the DataGraph CLI versus the DataGraph UI
```

= Using Anypoint Platform DataGraph CLI and API Extensions

You can use the Anypoint CLI DataGraph commands to add and edit API sources in DataGraph. To do so you must first add the appropriate DataGraph API extensions to the API source. 

== Adding Extensions to Your Project

> The Datagraph API extensions are available as a public asset in Exchange[link???]. You must add the extensions to your API-specification project as a dependency, and then publish the project to Exchange. After you publish your API-specification project, you can add the API to DataGraph using the CLI.

Author: @jisoldi  can we put a link where these will be published in exchange?

> The Datagraph API extensions are available as a public asset in Exchange[link???]. You must add the extensions to your API-specification project as a dependency, and then publish the project to Exchange. After you publish your API-specification project, you can add the API to DataGraph using the CLI.

Reviewer 2: The extension can't be viewed in Exchange (hence the lack of a link) and is only available as a fragment in API Designer.

Author: ```suggestion
You must add the extensions to your API-specification project as a dependency, and then publish the project to Exchange. After you publish your API-specification project, you can add the API to DataGraph using the CLI.
```

Author: @Extor3D is this correct then?

API Designer adds the required dependencies to `exchange.json` and validates the syntax of any annotations you added before publishing your project to Exchange.

== Maintaining RAML Compatibility

To maintain compatibility in RAML specifications with tools that do not support API extensions, these extensions includes a “RAML companion” that provides a simple version of RAML annotations definition. To use DataGraph API extensions in RAML, add the following code to the main file of the API-project specification:

----
uses:
  dg: exchange_modules/<group>/<asset>/<version>/datagraph-api-extensions.raml 
----

You can change the prefix `dg` if needed, but it must match the prefix you use the one use in the extensions.

The `<group>`, `<asset>`, and `<version>` reference the API extensions that are published in Exchange.

== Using the DataGraph CLI vs. the DataGraph UI

If you add an API source using the CLI, you cannot edit it or update its version directly in the UI. You must make these changes using the CLI. 

> //q: is the same in the other direction?  If you add an API using the UI can you update it later with the CLI?

Author: @jisoldi If you add an API using the UI can you update it later with the CLI?

Author: Got it, thanks
